extensions/wikia/SpecialBatchUserRights/SpecialBatchUserRights.php

```diff
@@ -42,6 +49,17 @@ public function __construct() {
 		parent::__construct( 'BatchUserRights' );
 	}
 
+	/**
+	 * @return UserPermissions
+	 */
+	private function userPermissions() {
+		if ( is_null( $this->permissionsService ) ) {
+			$this->permissionsService = Injector::getInjector()->get( PermissionsService::class );
+		}
+
+		return $this->permissionsService;
+	}
+
```

Comment:
This one is repeated at quite a few places. Let's consider introducing a trait that will provide `userPermissions` method.